apps/price_pusher/src/evm/evm.ts

```diff
@@ -362,51 +505,42 @@ export class PythContractFactory {
    *
    * @returns Pyth contract
    */
-  createPythContract(): Contract {
-    const provider = this.createWeb3Provider();
-    const web3 = new Web3(provider);
-    return new web3.eth.Contract(
-      AbstractPythAbi as any,
-      this.pythContractAddress
-    );
+  createPythContract(): PythContract {
+    return getContract({
+      address: this.pythContractAddress,
+      abi: PythAbi,
+      client: this.createPublicClient(),
+    });
   }
 
   hasWebsocketProvider(): boolean {
     return isWsEndpoint(this.endpoint);
   }
 
-  createWeb3Provider(): HttpProvider | WebsocketProvider {
-    if (isWsEndpoint(this.endpoint)) {
-      Web3.providers.WebsocketProvider.prototype.sendAsync =
-        Web3.providers.WebsocketProvider.prototype.send;
-      return new Web3.providers.WebsocketProvider(this.endpoint, {
-        clientConfig: {
-          keepalive: true,
-          keepaliveInterval: 30000,
-        },
-        reconnect: {
-          auto: true,
-          delay: 1000,
-          onTimeout: true,
-        },
-        timeout: 30000,
-      });
-    } else {
-      Web3.providers.HttpProvider.prototype.sendAsync =
-        Web3.providers.HttpProvider.prototype.send;
-      return new Web3.providers.HttpProvider(this.endpoint, {
-        keepAlive: true,
-        timeout: 30000,
-      });
-    }
+  createPublicClient(): PublicClient {
+    return createPublicClient({
+      transport: PythContractFactory.getTransport(this.endpoint),
+      chain: getChain(this.chainId),
+    });
   }
 
-  createWeb3PayerProvider() {
-    return new HDWalletProvider({
-      mnemonic: {
-        phrase: this.mnemonic,
-      },
-      providerOrUrl: this.createWeb3Provider() as ProviderOrUrl,
+  getAccount(): Account {
+    return mnemonicToAccount(this.mnemonic);
+  }
+
+  createWalletClient(): WalletClient {
+    return createWalletClient({
```

Comment:
You can also extend your Wallet Client with Public Actions if you just want to use a single Client (and not define both a Wallet & Public Client): https://viem.sh/docs/clients/wallet#optional-extend-with-public-actions

Reply:
The Public vs. Wallet Client separation is only really useful if you're interfacing with a JSON-RPC Account via an EIP-1193 Provider (ie. `window.ethereum`). This is because some of these Wallets may not expose some "public" RPC methods via their EIP-1193 Provider, so you'll also need to interface with a HTTP/WS RPC Provider as well. 

On the otherhand, if you're only using a Local Account like here (Mnemonic, PK, etc), then you don't run into these issues as you're always interfacing with a singular HTTP/WS RPC Provider.